app/goal_routes.py

```diff
@@ -0,0 +1,156 @@
+from flask import Blueprint, jsonify, abort, make_response, request
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+def validate_complete_request(request_body):
+    try:
+        if request_body["title"]:
+            return request_body
+
+    except:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+
+def validate_model_id(cls, model_id):
+    try:
+        model_id = int(model_id)    
+    except:
+        abort(make_response({"details": "Invalid data"}, 404))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"details": "Invalid data"}, 404))
+    
+    return model
+
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    valid_data = validate_complete_request(request_body)
+    new_goal = Goal.from_dict(valid_data)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    goal_response = {
+        "goal": new_goal.to_dict()
+    }
+    return make_response(jsonify(goal_response), 201)
+
+
+@goals_bp.route("", methods=["GET"])
+def get_all_goals_sort_asc():
+    goal_query = Goal.query.all()
+    title_query = request.args.get("title")
+    if title_query:
+        goal_query = Goal.query.filter_by(title=title_query)
+
+    goals = goal_query
+
+    goals_response = [goal.to_dict() for goal in goals]
+
+    return make_response(jsonify(goals_response), 200)
+
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    goal = validate_model_id(Goal, goal_id)
+
+    goal_response = {
+        "goal": goal.to_dict()
+    }
+
+    return make_response(jsonify(goal_response), 200)
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def goal_update_entire_entry(goal_id):
+    goal = validate_model_id(Goal, goal_id)
+    request_body = request.get_json()
+    goal.title = request_body["title"]
```

Comment:
You could run into an error on line 74 if a request was sent without a key "title". You'll want to have logic in PUT or PATCH methods (any route that takes in a request body) to guard against invalid data like you did in your POST request on line 34